File: src/main/java/dev/conorthedev/mediamod/media/spotify/api/track/Track.java
package dev.conorthedev.mediamod.media.spotify.api.track;

import dev.conorthedev.mediamod.media.spotify.api.album.Album;

/**
 * The implementation for a track in the Spotify API
 *
 * @author ConorTheDev
 */
public class Track {
    public Album album;
    public int duration_ms;
    public String name;
}

File: src/main/kotlin/dev/cbyrne/mediamod/services/media/Track.kt
package dev.cbyrne.mediamod.services.media

/*interface Album {

}

interface TrackInfo {
    var album: Album
    var progress_ms
}

interface Track {
    var progress: Int
    var playing: Boolean
}
 */

data class AlbumImage(val url: String)
data class Artist(val name: String)
data class Album(val artists: List<Artist>, val images: List<AlbumImage>, val name: String)
data class TrackInfo(val album: Album, val duration: Int, val name: String)
data class Track(val progress: Int, val playing: Boolean, val info: TrackInfo)